packages/grid/x-data-grid-pro/src/utils/domUtils.ts

```diff
@@ -10,6 +10,10 @@ export function findHeaderElementFromField(elem: Element, field: string): Elemen
   return elem.querySelector(`[data-field="${field}"]`);
 }
 
+export function findGroupHeaderElementsFromField(elem: Element, field: string): Element[] {
+  return Array.from(elem.querySelectorAll<HTMLDivElement>(`[data-fields~="|-${field}-|"]`) ?? []);
```

Comment:
It's not resizing the columns in real time. I guess you need to use `*=` here. https://stackoverflow.com/questions/28309119/css-difference-between-attribute-selectors-with-tilde-and-star

```suggestion
  return Array.from(elem.querySelectorAll<HTMLDivElement>(`[data-fields*="|-${field}-|"]`) ?? []);
```